packages/app-desktop/app.ts

```diff
@@ -428,6 +429,21 @@ class Application extends BaseApplication {
 
 		PerFolderSortOrderService.initialize();
 
+		async function createAllNoteFolder() {
+			const newFolder: FolderEntity = {
+				title: 'All Notes',
+			};
+			await Folder.save(newFolder, { userSideValidation: true });
+		}
+
+		async function checkAndCreateAllNoteFolder() {
+			const folderExists = await Folder.folderExists('All Notes');
+			if (!folderExists) {
+				await createAllNoteFolder();
+			}
+		}
+		await checkAndCreateAllNoteFolder();
+
```

Comment:
That's not right, whatever solution we go for we're not going to create an actual folder for this. So you need to make it work with the virtual folder as before

Reply:
Hi laurent22, I got it working with the virtual folder! Here's the video:

https://github.com/laurent22/joplin/assets/32552798/7ebc7855-57ff-4e7b-807e-a844c037e9e0

Reply:
I also updated the description of this PR:

- I made a virtual folder with ALL_NOTES_FILTER_ID in Sidebar.tsx. I chose to use the id because it's unique.
- 'All Notes' is now a FolderEntity, this decision was made due to easy integration with PerFolderSortOrderService.ts and a command 'toggleNotesSortOrderField'.
- the event was not listening on the 'All Notes' so I refactored the getSelectedFolderId() function.

Reply:
I had a second look and made the solution much cleaner. With understanding the codebase better, I added `onClick={() => { onAllNotesClick_; folderItem_click(ALL_NOTES_FILTER_ID); }}`, so the event is listening now on the 'All Notes' folder.

Only changes are made in Sidebar.tsx now.

Reply:
Let me know whether using `ALL_NOTES_FILTER_ID` is allowed and a good solution. 

I wanted to assign an ID for 'All Notes' that would not cause conflicts with existing or new folder IDs. That is why I chose `ALL_NOTES_FILTER_ID`.